### PR TITLE
Cherry-pick #7285 to 6.3: permit index-pattern only setup

### PR DESCRIPTION
**Pull request by ruflin**
Cherry-pick of PR #7285 to 6.3 branch. Original message: 

with `setup.dashboards.enabled=true`, `setup.dashboards.only_index=true` always throws
"Error importing Kibana dashboards: fail to import the dashboards in Kibana"

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -125,6 +125,8 @@ https://github.com/elastic/beats/compare/v6.2.3...v6.3.0[View commits]
 - Fix map overwrite panics by cloning shared structs before doing the update. {pull}6947[6947]
 - Fix delays on autodiscovery events handling caused by blocking runner stops. {pull}7170[7170]
 - Do not emit Kubernetes autodiscover events for Pods without IP address. {pull}7235[7235]
+- Allow to override the `ignore_above` option when defining new field with the type keyword. {pull}7238[7238]
```
Comment (ph) [inline CHANGELOG.asciidoc]: The change is not part of this PR and this line should not be present.

Comment (ruflin) [reply CHANGELOG.asciidoc]: fixed

**Comment by ph** [comment]
failures is unrelated to this change.
